massa-ledger-exports/src/ledger_changes.rs

```diff
@@ -20,38 +20,24 @@ use nom::error::{context, ContextError, ParseError};
 use nom::multi::length_count;
 use nom::sequence::tuple;
 use nom::{IResult, Parser};
-use serde::{ser::SerializeSeq, Deserialize, Serialize};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
 use std::collections::{hash_map, BTreeMap};
 use std::ops::Bound::Included;
 
 /// represents an update to one or more fields of a `LedgerEntry`
+#[serde_as]
 #[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct LedgerEntryUpdate {
     /// change the balance
     pub balance: SetOrKeep<Amount>,
     /// change the executable bytecode
     pub bytecode: SetOrKeep<Bytecode>,
     /// change datastore entries
-    #[serde(serialize_with = "as_array")]
+    #[serde_as(as = "Vec<(_, _)>")]
     pub datastore: BTreeMap<Vec<u8>, SetOrDelete<Vec<u8>>>,
```

Comment:
Where do we use serde serialization for this struct? Would it not be better to implement our own (de)serializer?

Reply:
he has been added here : https://github.com/massalabs/massa/pull/4332

Fix also : https://github.com/massalabs/massa/issues/4428

Reply:
Ok thanks!